bin/varnishd/mgt/mgt_main.c

```diff
@@ -881,6 +882,13 @@ main(int argc, char * const *argv)
 		    workdir, VAS_errtxt(errno));
 	}
 
+	if (VJ_make_subdir("worker_tmpdir",
+	    "TMPDIR for the worker process", NULL)) {
+		ARGV_ERR(
+		    "Cannot create vmod directory (%s/worker_tmpdir): %s\n",
```

Comment:
Is this limited to VMODs? Are you assuming enough privileges in the first place for VEXTs?

Reply:
Should Varnish provide helper function to create the temp files to avoid collision between the consumer of this temp directory?

Reply:
We should at least provide a macro or a `const char *` (for actual linking) with the directory name and let VMODs be responsible for their own stuff. Not sure I'd want to reinvent `mktemp` for VMODs... :shrug:

Reply:
The proposed directory is for any code executing within the worker process, hence the name.
I am 🤷🏽 on the macro/string also, because it would just be `worker_tmpdir` relative to cwd.

Reply:
Likewise, I approved it as it is.